app/models/chargeback_rate_detail.rb

```diff
@@ -90,4 +90,26 @@ def rate_type
     # Return parent's rate type
     chargeback_rate.rate_type unless chargeback_rate.nil?
   end
+
+  def detail_currency
+    ChargebackRateDetailCurrency.find_by(:id => chargeback_rate_detail_currency_id)
+  end
+
+  # New method created in order to show the rates in a easier to understand way
+  def show_rates
+    rate = self.rate.to_s
+    return detail_currency.code if rate.to_f.zero?
+    hr = case per_time
```

Comment:
You could probably simplify this by putting the mapping per_time to the friendly string in a hash and doing a lookup. Something like this:
```
    per_time_map = {
      :hourly  =>"Hour",
      :daily   => "Day",
      :weekly  => "Week",
      :monthly => "Month",
      :yearly  => "Year"
    }
    hr = per_time_map[per_time.to_sym]
    raise "rate time unit of '#{per_time}' not supported" if hr.nil?
```
Another option would be to add them to the message file config/locales/en.yml and do the lookup there. This approach would make it easier to internationalize the strings.